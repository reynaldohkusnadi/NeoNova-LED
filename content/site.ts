/**
 * Authoring workflow:
 * - Edit this file in a feature branch via PR.
 * - Preview will deploy and render content from this file only (no hardcoded strings in components).
 * - Once approved, merge to main; Production reads the same typed export.
 */
export interface Card {
  id: string;
  headline: string;
  blurbShort: string;
  ctaLabel: string;
}

export interface Stat {
  label: string;
  value: string;
}

export interface Content {
  headline: string;
  blurbShort: string;
  cardsCore: [Card, Card, Card];
  stats: [Stat, Stat];
  form: {
    success: string;
    ctaPrimary: string;
    ctaSecondary: string;
    whatsappPrefill: string;
  };
}

export const content = {
  headline: "Neo Nova: Accelerate product impact",
  blurbShort:
    "We build high‑performance web experiences with motion, 3D, and analytics baked in.",
  cardsCore: [
    {
      id: "speed",
      headline: "Blazing performance",
      blurbShort: "Ship fast experiences that convert.",
      ctaLabel: "See results",
    },
    {
      id: "motion",
      headline: "Meaningful motion",
      blurbShort: "Enhance clarity with delightful motion.",
      ctaLabel: "View work",
    },
    {
      id: "quality",
      headline: "Quality guardrails",
      blurbShort: "CI gates for accessibility and speed.",
      ctaLabel: "How it works",
    },
  ],
  stats: [
    { label: "Lighthouse", value: "90+" },
    { label: "Axe criticals", value: "0" },
  ],
  form: {
    success: "Thanks! We'll be in touch shortly.",
    ctaPrimary: "Request demo",
    ctaSecondary: "Chat on WhatsApp",
    whatsappPrefill: "Hi Neo Nova team — I'd love to learn more.",
  },
} as const satisfies Content;


