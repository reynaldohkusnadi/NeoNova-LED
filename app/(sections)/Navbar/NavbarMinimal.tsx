import React from "react";
import { content } from "@/content/site";

export default function NavbarMinimal() {
  const { ctaPrimary, ctaSecondary } = content.form;
  return (
    <header
      className="sticky top-0 z-[var(--nn-z-header)] h-16 flex items-center border-b bg-[color-mix(in_oklab,var(--background)_92%,black)] backdrop-blur supports-[backdrop-filter]:bg-[color-mix(in_oklab,var(--background)_85%,black)]"
    >
      <nav
        className="w-full max-w-6xl mx-auto px-4 flex items-center justify-between"
        aria-label="Primary"
      >
        <a href="#" className="font-medium">Neo Nova</a>
        <div className="flex gap-3 items-center">
          <a
            href="#"
            data-analytics-event="cta_click_secondary"
            className="px-3 py-2 rounded-md border"
          >
            {ctaSecondary}
          </a>
          <a
            href="#"
            data-analytics-event="cta_click_primary"
            className="px-3 py-2 rounded-md bg-[var(--nn-foreground)] text-[var(--nn-primary-contrast)] dark:bg-white dark:text-[var(--nn-foreground)]"
          >
            {ctaPrimary}
          </a>
        </div>
      </nav>
    </header>
  );
}


