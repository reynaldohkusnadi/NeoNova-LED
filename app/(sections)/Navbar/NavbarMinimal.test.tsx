import React from "react";
import { render, screen } from "@testing-library/react";
import NavbarMinimal from "./NavbarMinimal";
import { content } from "@/content/site";

describe("NavbarMinimal", () => {
  it("renders CTA labels from content", () => {
    const { container } = render(<NavbarMinimal />);
    expect(screen.getByText(content.form.ctaPrimary)).toBeInTheDocument();
    expect(screen.getByText(content.form.ctaSecondary)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});


