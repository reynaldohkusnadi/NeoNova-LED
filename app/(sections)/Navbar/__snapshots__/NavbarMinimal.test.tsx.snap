// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavbarMinimal > renders CTA labels from content 1`] = `
<header
  class="sticky top-0 z-[var(--nn-z-header)] h-16 flex items-center border-b bg-[color-mix(in_oklab,var(--background)_92%,black)] backdrop-blur supports-[backdrop-filter]:bg-[color-mix(in_oklab,var(--background)_85%,black)]"
>
  <nav
    aria-label="Primary"
    class="w-full max-w-6xl mx-auto px-4 flex items-center justify-between"
  >
    <a
      class="font-medium"
      href="#"
    >
      Neo Nova
    </a>
    <div
      class="flex gap-3 items-center"
    >
      <a
        class="px-3 py-2 rounded-md border"
        data-analytics-event="cta_click_secondary"
        href="#"
      >
        Chat on WhatsApp
      </a>
      <a
        class="px-3 py-2 rounded-md bg-[var(--nn-foreground)] text-[var(--nn-primary-contrast)] dark:bg-white dark:text-[var(--nn-foreground)]"
        data-analytics-event="cta_click_primary"
        href="#"
      >
        Request demo
      </a>
    </div>
  </nav>
</header>
`;
